packages/block-editor/src/hooks/layout.native.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import { removeFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import './layout.js';
+
+// This filter is removed because layout styles shouldn't be added
+// until layout types are supported in the native version.
+removeFilter(
+	'editor.BlockListBlock',
+	'core/editor/layout/with-layout-styles'
+);
+
+// This filter is removed because the layout controls shouldn't be
+// enabled until layout types are supported in the native version.
+removeFilter(
+	'editor.BlockEdit',
+	'core/editor/layout/with-inspector-controls'
+);
```

Comment:
Here you can see the logic related to this hook:
https://github.com/WordPress/gutenberg/blob/29ec9e0261c1c915bb75db4c4679f91225f685d1/packages/block-editor/src/hooks/layout.js#L160-L181

Reply:
In the Buttons block, regarding the justification content controls, instead of using the one provided by the layout logic:
https://github.com/WordPress/gutenberg/blob/29ec9e0261c1c915bb75db4c4679f91225f685d1/packages/block-editor/src/layouts/flex.js#L69-L86

We define it within the block render:
https://github.com/WordPress/gutenberg/blob/29ec9e0261c1c915bb75db4c4679f91225f685d1/packages/block-library/src/buttons/edit.native.js#L126-L138

Reply:
Not suggesting a change here, but I'm wondering whether there is an opportunity to extract some of what is in the web implementation. Certainly not a blocker for this PR, just thinking about a possibility where `FlexLayoutJustifyContentControl` was extracted, if that would highlight where the platforms deviate. It may be that it isn't worth trying to merge components at that level, if it is mostly UI, but I'm thinking more for the sake of the logical parts of it, like the `allowedControls`, etc. :thinking: